atomkraft/chain/utils.py

```diff
@@ -46,7 +46,11 @@ async def _wait(self):
         while True:
             msg = responses.to_result(json.loads(await self.websocket.recv()))
             if "query" in msg.result and msg.result["query"] == self.query:
-                break
+                try:
+                    if self.filter(msg.result):
```

Comment:
I am a bit confused by this line: where is the member function `filter` defined?

Reply:
I added a `set_filter` method.